pkg/metadata/store.go

```diff
@@ -28,22 +37,48 @@ func SyncMetadata(ctx context.Context, first storage.Backend, second storage.Bac
 
 // UpdateMetadata runs some reconciliation functions on Metadata to ensure its state is consistent
 // then uses the Backend to update the metadata storage medium.
-func UpdateMetadata(ctx context.Context, backend storage.Backend, meta *v1alpha2.Metadata, skipTLSVerify, plainHTTP bool) error {
-
-	// If using heads only setting, aggregate all of the associations.
-	// Replace if using ranges to prune any old image information
-	// to support pruning.
-	if keepPastAssociations(meta.PastMirror.Mirror) {
-		meta.PastAssociations = append(meta.PastAssociations, meta.PastMirror.Associations...)
-	} else {
-		meta.PastAssociations = meta.PastMirror.Associations
+func UpdateMetadata(ctx context.Context, backend storage.Backend, meta *v1alpha2.Metadata, workspace string, skipTLSVerify, plainHTTP bool) error {
+	pastMeta := v1alpha2.NewMetadata()
+	pastReleases := map[string]string{}
+	merr := backend.ReadMetadata(ctx, &pastMeta, config.MetadataBasePath)
+	if merr != nil && !errors.Is(merr, storage.ErrMetadataNotExist) {
+		return merr
+	} else if merr == nil {
+		for _, ch := range pastMeta.PastMirror.Platforms {
+			pastReleases[ch.ReleaseChannel] = ch.MinVersion
+		}
 	}
 
+	mirror := meta.PastMirror
+	// Store starting versions for new catalogs
+	logrus.Debugf("Resolving operator metadata")
 	var operatorErrs []error
 
-	mirror := meta.PastMirror
+	resolver, err := containerdregistry.NewResolver("", skipTLSVerify, plainHTTP, nil)
+	if err != nil {
+		return fmt.Errorf("error creating image resolver: %v", err)
+	}
+	cacheDir, err := os.MkdirTemp("", "imageset-catalog-registry-")
+	if err != nil {
+		return err
+	}
+
+	logger := logrus.New()
```

Comment:
Interesting. Look like this is just a placeholder log for the containderd stuff. Is there an option not to include the log option in the `NewRegistry()` method?

Reply:
I can add a comment here to clarify. We only use this to set the output to `ioutil.Discard` because the `containerd.Registry` was a bit verbose.